bundles/org.openhab.binding.nobohub/src/main/java/org/openhab/binding/nobohub/internal/NoboHubBridgeHandler.java

```diff
@@ -0,0 +1,432 @@
+                    try {
+                        OverrideMode mode = OverrideMode.getByName(strCommand.toFullString());
+                        ht.getConnection().setOverride(h, mode);
+                    } catch (NoboCommunicationException nce) {
+                        logger.error("Failed setting override mode", nce);
+                    } catch (NoboDataException nde) {
+                        logger.error("Date format error setting override mode", nde);
+                    }
+                } else {
+                    logger.error("Command of wrong type: {} ({})", command, command.getClass().getName());
+                }
+            } else {
+                if (null == hub) {
+                    logger.error("Could not set override, hub not detected yet");
+                }
+
+                if (null == hubThread) {
+                    logger.error("Could not set override, hub connection thread not set up yet");
+                }
+            }
+        }
+    }
+
+    @java.lang.Override
+    public void initialize() {
+        config = getConfigAs(NoboHubBridgeConfiguration.class);
+
+        if (null == config) {
+            logger.error("Missing Configuration");
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "No configuration set");
+            return;
+        }
+
+        NoboHubBridgeConfiguration c = Helpers.castToNonNull(config, "config");
+
+        String serialNumber = c.serialNumber;
+        if (null == serialNumber) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "Missing serial number in configuration");
+            return;
+        }
+
+        String hostName = c.hostName;
+        if (null == hostName) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "Missing host name in configuration");
+            return;
+        }
+
+        logger.info("Looking for Hub {} at {}", c.serialNumber, c.hostName);
+
+        // Set the thing status to UNKNOWN temporarily and let the background task decide for the real status.
+        updateStatus(ThingStatus.UNKNOWN);
+
+        // Background handshake:
+        scheduler.execute(() -> {
+            try {
+                HubConnection conn = new HubConnection(hostName, serialNumber, this);
+                conn.connect();
+
+                logger.debug("Done connecting to {} ({})", hostName, serialNumber);
+
+                Duration timeout = Duration.ofSeconds(14);
+                if (c.pollingInterval > 0) {
+                    timeout = Duration.ofSeconds(c.pollingInterval);
+                }
+
+                logger.debug("Starting communication thread to {}", hostName);
+
+                HubCommunicationThread ht = new HubCommunicationThread(conn, timeout);
+                ht.start();
+                hubThread = ht;
+
+                if (ht.getConnection().isConnected()) {
```

Comment:
If the network is permanent/temporarily down during initial connection or the IP address is wrong, this will fail and return "Error while connecting: Failed connecting to Nobø Hub at A.B.C.D" in the UI. I think there is potential for improvement here, but the solution probably needs a little bit of work. The current retry mechanism in the HubCommunicationThread.class works quiet well, but does have two disadvantage, it will never start if the initial connection fails and it will fail if there is a medium long network outage (30 seconds+), resulting in the connection to the Hub stopping completely, without any future retry. Looking at some of the other BrigdeHandlers in the repo, I see some are setting up ScheduledFuture polling jobs. It this the recommended approach? And is it OK to let such an mechanism retry forever?

Reply:
Yes, that's an often used approach and completely acceptable.

Reply:
We have started looking into improving this, but it will take some time to make a stable solution. I suggest we add that at a later time. The current solution does have it risk, but it`s a low one.

Reply:
OK. You could have a look at #11891, not sure if it will be usable here, but this was a simple approach to solving a similar problem, IIRC.